source/message.h

```diff
@@ -42,6 +44,25 @@ using MessageConsumer = std::function<void(
     const spv_position_t& /* position */, const char* /* message */
     )>;
 
+// A message consumer that ignores all messages.
+inline void IgnoreMessage(MessageLevel, const char*, const spv_position_t&,
+                          const char*) {}
+
+// A helper function to compose and return a string from the message in the
+// following format:
+//   "<level>: <source>:<line>:<column>:<index>: <message>"
+std::string StringifyMessage(MessageLevel level, const char* source,
+                             const spv_position_t& position,
+                             const char* message);
+
+// Prints the given message to stderr following the format in
+// StringifyMessage().
+inline void PrintMessageToStderr(MessageLevel level, const char* source,
+                                 const spv_position_t& position,
+                                 const char* message) {
+  std::cerr << StringifyMessage(level, source, position, message);
```

Comment:
Should hide this body.  Then you won't need to #include iostream and make client compilation slower.
Also, I suggest having the following more general thing:
```
inline MessageConsumer MessagePrinterToStream(std::ostream* out) { 
 return [out](MessageLevel level, const char* source, const spv_position_t& position, const char* message) { 
    *out << StringifyMessage(level, source, position, message);
};
}
```
I think you shouldn't provide PrintMessageToStderr once you have the more general thing.  In a real program you'll only see the std::cerr case once, in the main.cpp file.